### PR TITLE
CLI to failure when no authorization key configured/provided (prior to sending HTTP request)

### LINKED ISSUE #778: Go CLI Should Not Send Request if Auth Key is Not Set
Currently the Go CLI sends request if an auth key is not set. This request is rejected by the Whisk backend, and an error is displayed to the user. This logic should be reworked so that the Go CLI does not send the request if the auth key is not set, and display an error to the user.

Tests must be updated to handle this change.

See comments: https://github.com/openwhisk/openwhisk/pull/614#discussion-diff-68622642

### PR DESCRIPTION
**Pull request by mdeuser**
Check that the auth key is set prior to sending the HTTP request
Fail the CLI if the require auth key is not set
Remove auth key from `sdk install` as it is not needed
Refactor the namespace code to use the common HTTP client code
Fix incorrect argument help text
Fixes #778

### CODE REVIEW

**Comment by mdeuser** [comment]
@dubeejw - Ready for review.  Thanks!

**Comment by mdeuser** [comment]
@dubeejw - rebased against master branch; openwhisk CLI tests all pass

**Comment by dubeejw** [comment]
@mdeuser, do all the tests pass when you flip the switch to use the Go CLI in WskBasicTests.scala?

**Comment by dubeejw** [comment]
Also it looks like there are other changes in here that don't relate to the auth key issue?

**Comment by mdeuser** [comment]
It looks like my comments in the commit all disappeared during the rebase.  The only change not related to the auth check is a change to the property command help text that I noticed was incorrect while using `wsk property` in testing wsk without any --auth setting.

**Comment by mdeuser** [comment]
@dubeejw - Good point.  All 400+ tests for both python and Go CLI pass in my local env.

**Comment by mdeuser** [comment]
@dubeejw - What exactly is the way to flip the test switch (Python -> Go CLI)?  I thought I changed it correctly, but now I'm suspicious.  Let's Hold off on merge until we have a chance to talk tomorrow.

**Comment by mdeuser** [comment]
PG 228.
Added back code in WskBasicTests.scala to propogate the usePythonCLI attribute into the other command specific classes.  That fixed the issue I was seeing with the CLI switching from GO to Python mid execution of WskBasicTests.  Tests all now pass locally with Python and Go enabled in WskBasicTests.
@dubeejw - ready for re-review.

**Comment by dubeejw** [comment]
@mdeuser, build failed due to trailing whitespace.

File: tests/src/common/Wsk.scala [hunk 1]
```diff
@@ -641,11 +646,12 @@ class WskNamespace()
     }
 }
 
-class WskPackage()
+class WskPackage(override val usePythonCLI: Boolean = true) 
     extends RunWskCmd
     with ListOrGetFromCollection
     with DeleteFromCollection {
     override protected val noun = "package"
+    override def baseCommand = Wsk.baseCommand(usePythonCLI)
```
Comment (rabbah) [inline tests/src/common/Wsk.scala]: I suggest also making the sealed trait's property abstract (that way there can't be confusion)

```
diff --git a/tests/src/common/Wsk.scala b/tests/src/common/Wsk.scala
index 08f3bba..a3a3b6c 100644
--- a/tests/src/common/Wsk.scala
+++ b/tests/src/common/Wsk.scala
@@ -787,7 +787,7 @@ object Wsk {
 
 sealed trait RunWskCmd {
 
-    val usePythonCLI: Boolean = true
+    val usePythonCLI: Boolean
 
```

File: tests/src/common/Wsk.scala [hunk 2]
```diff
@@ -202,13 +202,14 @@ trait HasActivation {
     }
 }
 
-class WskAction()
+class WskAction(override val usePythonCLI: Boolean = true)
```
Comment (dubeejw) [inline tests/src/common/Wsk.scala]: Trailing whitespace.

File: tests/src/common/Wsk.scala [hunk 3]
```diff
@@ -267,13 +268,14 @@ class WskAction()
     }
 }
 
-class WskTrigger()
+class WskTrigger(override val usePythonCLI: Boolean = true)
```
Comment (dubeejw) [inline tests/src/common/Wsk.scala]: Trailing whitespace.

File: tests/src/common/Wsk.scala [hunk 4]
```diff
@@ -317,13 +319,14 @@ class WskTrigger()
     }
 }
 
-class WskRule()
+class WskRule(override val usePythonCLI: Boolean = true)
```
Comment (dubeejw) [inline tests/src/common/Wsk.scala]: Trailing whitespace.

File: tests/src/common/Wsk.scala [hunk 5]
```diff
@@ -432,12 +435,13 @@ class WskRule()
     }
 }
 
-class WskActivation()
+class WskActivation(override val usePythonCLI: Boolean = true)
```
Comment (dubeejw) [inline tests/src/common/Wsk.scala]: Trailing whitespace.

File: tests/src/common/Wsk.scala [hunk 6]
```diff
@@ -608,11 +612,12 @@ class WskActivation()
     private case class PartialResult(ids: Seq[String]) extends Throwable
 }
 
-class WskNamespace()
+class WskNamespace(override val usePythonCLI: Boolean = true)
```
Comment (dubeejw) [inline tests/src/common/Wsk.scala]: Trailing whitespace.

File: tests/src/common/Wsk.scala [hunk 7]
```diff
@@ -641,11 +646,12 @@ class WskNamespace()
     }
 }
 
-class WskPackage()
+class WskPackage(override val usePythonCLI: Boolean = true)
```
Comment (dubeejw) [inline tests/src/common/Wsk.scala]: Trailing whitespace.

File: tests/src/common/Wsk.scala (hunk 2 above)
Comment (rabbah) [inline tests/src/common/Wsk.scala]: @csantanapr we should document the git commit hook with runs scan code for devs to adopt.

**Comment by mdeuser** [comment]
Travis is now happy.  Since there was a rebase, a new PG 264 is being run

**Pull request merged**